projects/wave-core-new/src/lib/layers/symbology/symbology.model.ts

```diff
@@ -35,711 +49,733 @@ export const MIN_ALLOWED_POINT_RADIUS = 1;
-        switch (dict.symbologyType) {
-            case SymbologyType[SymbologyType.SIMPLE_POINT]:
-            case SymbologyType[SymbologyType.COMPLEX_POINT]:
-                return PointSymbology.createSymbology(dict as PointSymbologyDict);
-
-            case SymbologyType[SymbologyType.CLUSTERED_POINT]:
-                return PointSymbology.createClusterSymbology(dict as VectorSymbologyDict);
-
-            case SymbologyType[SymbologyType.SIMPLE_LINE]:
-            case SymbologyType[SymbologyType.COMPLEX_LINE]:
-                return LineSymbology.createSymbology(dict as VectorSymbologyDict);
-
-            case SymbologyType[SymbologyType.SIMPLE_VECTOR]:
-            case SymbologyType[SymbologyType.COMPLEX_VECTOR]:
-                return VectorSymbology.createSymbology(dict as VectorSymbologyDict);
-
-            case SymbologyType[SymbologyType.RASTER]:
-            case SymbologyType[SymbologyType.MAPPING_COLORIZER_RASTER]:
-                return MappingRasterSymbology.createSymbology(dict as RasterSymbologyDict);
-            default:
-                throw new Error('Unsupported AbstractSymbology');
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IconStyle {}
+
+export abstract class Symbology {
+    static fromDict(dict: SymbologyDict): Symbology {
+        if (dict.Raster) {
+            return RasterSymbology.fromRasterSymbologyDict(dict.Raster);
+        } else if (dict.Vector) {
+            const vectorDict = dict.Vector;
+            if (vectorDict.Point) {
+                return PointSymbology.fromPointSymbologyDict(vectorDict.Point);
+            } else if (vectorDict.Line) {
+                return LineSymbology.fromLineSymbologyDict(vectorDict.Line);
+            } else if (vectorDict.Polygon) {
+                return PolygonSymbology.fromPolygonSymbologyDict(vectorDict.Polygon);
+            }
         }
+        throw new Error('Invalid Symbology type.');
     }
 
+    abstract toDict(): SymbologyDict;
+
+    abstract clone(): Symbology;
+
+    abstract equals(other: Symbology): boolean;
+
     abstract getSymbologyType(): SymbologyType;
 
+    abstract getIconStyle(): IconStyle;
+
+    get iconStyle(): IconStyle {
+        return this.getIconStyle();
+    }
+
     get symbologyType(): SymbologyType {
         return this.getSymbologyType();
     }
+}
 
-    get symbologyTypeId(): string {
-        return SymbologyType[this.getSymbologyType()];
+export abstract class VectorSymbology extends Symbology {
+    createStyleFunction(): OlStyleFunction {
+        return (feature: OlFeature, _resolution: number): OlStyle => {
+            const styler = this.createStyler(feature);
+
+            const key = styler.cacheKey();
+            if (!(key in styleCache)) {
+                const style = styler.createStyle();
+                styleCache[key] = style;
+                return style;
+            } else {
+                return styleCache[key];
+            }
```

Comment:
why don't you just reverse the branches?

Reply:
449a8a34ffcfa29f49afba5b97a1188706887ac4